File: phpdev/vc15/x64/main.py
import datetime

def test(n, p):
    for i in range(0, n):
        for j in range(0, n):
            for k in range(0, n):
                if p and i+j+k==n and i*i+j*j==k*k:
                    p(i, j, k)

def test_print(a, b, c):
    print "The result %d^2 + %d^2 = %d^2 !" % (a, b, c)

def main():
    starttime = datetime.datetime.now()

    test(1000, test_print)
    print "First Used %f seconds" % ((datetime.datetime.now() - starttime).seconds, )

    for i in range(0, 10):
        test(1000, None)

    print "All Used %f seconds" % ((datetime.datetime.now() - starttime).seconds, )

if __name__ == '__main__':
    main()

File: phpdev/vc15/x64/main.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#if defined(__clang__)
    #define NO_INLINE __attribute__((noinline))
#else
    #define NO_INLINE
#endif

static void  test(int n, void (*p)(int, int, int)) 
NO_INLINE
{
    for (int i = 0; i <= n; i++)
    {
        for (int j = 0; j <= n; j++)
        {
            for (int k = 0; k <= n; k++)
            {
                if (p && i + j + k == n && i * i + j * j == k * k)
                {
                    p(i, j, k);
                }
            }
        }
    }
}

static void test_print(int a, int b, int c)
NO_INLINE
{
    printf("The result %d^2 + %d^2 = %d^2 !\n", a, b, c);
}

int main()
{
    clock_t start, finish;
    start = clock();
    double duration;

    test(1000, test_print);
    finish = clock();
    duration = (double)(finish - start) / CLOCKS_PER_SEC;
    printf("First Used %f seconds\n", duration);

    for (int i = 0; i <= 10; i++)
    {
        test(1000, NULL);
    }

    finish = clock();
    duration = (double)(finish - start) / CLOCKS_PER_SEC;
    printf("All Used %f seconds\n", duration);
    return 0;
}
